frame/society/src/lib.rs

```diff
@@ -823,8 +829,37 @@ decl_module! {
 			Self::add_member(&founder)?;
 			<Head<T, I>>::put(&founder);
 			<Founder<T, I>>::put(&founder);
+			Rules::<T, I>::put(T::Hashing::hash(&rules));
 			Self::deposit_event(RawEvent::Founded(founder));
 		}
+
+		/// Anull the founding of the society.
+		///
+		/// This may be done when there is only one member.
+		///
+		/// The dispatch origin for this call must be Signed, and the signing account must be the
+		/// `Founder`.
+		///
+		/// # <weight>
+		/// - One storage read, O(M).
+		/// - Four storage removals O(1).
+		/// - One event.
+		///
+		/// Total Complexity: O(1)
```

Comment:
Sorry, I probably lack context, but how `O(M)` in `One storage read` gives total `O(1)` here?

Reply:
`O(M)` above probably means the length of the storage value, right?

Reply:
yeah but you're right - that was a typo.